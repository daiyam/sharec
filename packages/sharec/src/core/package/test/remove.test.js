const { vol } = require('memfs')
const {
  clearPackageJson,
  ereaseDependencies,
  ereaseMetaData,
} = require('../remove')

describe('core > package > remove >', () => {
  const packageJson5 = require('fixtures/package/package_05.json')
  const packageJson6 = require('fixtures/package/package_06.json')
  const packageJsonFixture = require('fixtures/package/package_07.json')

  beforeEach(() => {
    vol.reset()
  })

  describe('clearPackageJson', () => {
    it('should remove configs from package json and injection status', async () => {
      const dir = {
        '/target/package.json': JSON.stringify(packageJson5),
        '/configuration-package/package.json': JSON.stringify(packageJson6),
      }

      vol.fromJSON(dir)

      await clearPackageJson('/configuration-package', '/target')

      expect(
        JSON.parse(vol.readFileSync('/target/package.json', 'utf8')),
      ).toMatchSnapshot()
    })
  })

  describe('ereaseDependencies', () => {
    it('should remove all injected configs in package.json from target path', () => {
      const deps = {
        devDependencies: packageJsonFixture.devDependencies,
      }
      const res = ereaseDependencies(deps)(packageJsonFixture)

      expect(res).toEqual([{ ...packageJsonFixture, devDependencies: {} }, {}])
    })

    it('should remove all injected configs except changed by user after injection', () => {
      const deps = {
        devDependencies: packageJsonFixture.devDependencies,
      }
      const modifiedDeps = {
        '@commitlint/cli': '^8.6.1',
        '@commitlint/config-conventional': '^8.6.0',
        'babel-eslint': '^11.0.1',
        commitizen: '^2.1.1',
      }
      const res = ereaseDependencies(deps)({
        ...packageJsonFixture,
        devDependencies: {
          ...packageJsonFixture.devDependencies,
          ...modifiedDeps,
        },
      })

      expect(res).toEqual([
        {
          ...packageJsonFixture,
          devDependencies: modifiedDeps,
        },
        {
          '@commitlint/cli': '^7.6.1 -> ^8.6.1',
          '@commitlint/config-conventional': '^7.6.0 -> ^8.6.0',
          'babel-eslint': '^10.0.1 -> ^11.0.1',
          commitizen: '^3.1.1 -> ^2.1.1',
        },
      ])
    })
  })

  describe('ereaseMetaData', () => {
    it('should remove all sharec meta-data in package.json from target path', () => {
      const res = ereaseMetaData({
        sharec: {
          injected: true,
        },
      })

      expect(res).toEqual({})
    })
  })
})
