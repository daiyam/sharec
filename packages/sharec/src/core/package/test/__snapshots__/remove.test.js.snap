// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core > package > remove > clearPackageJson should remove configs from package json and injection status 1`] = `
Object {
  "config": Object {
    "commitizen": Object {
      "path": "cz-conventional-changelog",
    },
  },
  "devDependencies": Object {
    "@commitlint/cli": "^7.6.1",
    "cz-conventional-changelog": "^2.1.0",
    "eslint-config-prettier": "^4.2.0",
  },
  "eslintConfig": Object {
    "env": Object {
      "node": false,
    },
    "extends": "standard",
    "globals": Object {
      "baz": true,
    },
    "rules": Object {
      "max-len": Array [
        "warn",
        80,
        2,
      ],
      "sort-imports": 0,
    },
  },
  "eslintIgnore": Array [
    "/node_modules",
  ],
  "husky": Object {
    "hooks": Object {
      "commit-msg": "commitlint -E",
      "pre-commit": "lint-staged",
    },
  },
  "lint-staged": Object {
    "src/**/*.{js,json}": Array [
      "eslint --fix",
      "prettier --write",
      "git add",
    ],
  },
  "name": "awesome-project",
  "prettier": Object {
    "semi": false,
    "singleQuote": true,
    "trailingComma": "all",
  },
  "scripts": Object {
    "commit": "git-cz",
    "lint": "eslint 'src/**/*.{jsx,json}'",
    "lint-staged": "lint-staged",
  },
}
`;
